File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0
            
        s = set()

        longest = 1
        curr = 1

        for num in nums:
            s.add(num)

        for n in s:
            if n - 1 not in s:
                i = 1
                while n + i in s:
                    curr += 1
                    i += 1
                longest = max(curr,longest)
                curr = 1
        return longest

File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_map<int, int> dupMap;

        for(int num: nums){
            if(dupMap.find(num) == dupMap.end()){
                dupMap[num] = 1;
            }else{
                dupMap[num] += 1;
            }
        }

        for(auto& pair: dupMap){
            if(pair.second > 1){
                return true;
            }
        }
        return false;
    }
};
